### PR TITLE
[344] Implement effectiveName for SysML and KerML elements

### PR DESCRIPTION
**Pull request by adaussy**
Bug: https://github.com/eclipse-syson/syson/issues/344

### CODE REVIEW

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/FeatureImpl.java [hunk 1]
```diff
@@ -699,13 +699,14 @@ public boolean isFeaturedWithin(Type type) {
...
      *
-     * @generated
+     * @generated not
      */
     @Override
     public Feature namingFeature() {
-        // TODO: implement this method
-        // Ensure that you remove @generated or mark it @generated NOT
-        return null;
+        return this.getOwnedRedefinition().stream()//
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/FeatureImpl.java:706]: Please remove // at the end of lines

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/FeatureImpl.java [hunk 2]
```diff
@@ -699,13 +699,14 @@ public boolean isFeaturedWithin(Type type) {
     /**
      * <!-- begin-user-doc --> <!-- end-user-doc -->
      *
-     * @generated
+     * @generated not
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/FeatureImpl.java:702]: not -> NOT

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/LifeClassImpl.java [hunk 3]
```diff
@@ -31,6 +31,11 @@ protected LifeClassImpl() {
         super();
     }
 
+    @Override
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/LifeClassImpl.java:34]: Please add @generated NOT

File: backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/ConjugatedPortDefinitionImplTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package org.eclipse.syson.sysml.impl;
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/ConjugatedPortDefinitionImplTest.java:1]: Please add copyright

File: backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/FeatureImplTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
+package org.eclipse.syson.sysml.impl;
```
Comment (AxelRICHARD) [inline backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/FeatureImplTest.java:1]: Please add copyright

Comment (adaussy) [reply backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/FeatureImplTest.java:1]: Done

File: backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/ConjugatedPortDefinitionImplTest.java (hunk 4 above)
Comment (adaussy) [reply backend/metamodel/syson-sysml-metamodel/src/test/java/org/eclipse/syson/sysml/impl/ConjugatedPortDefinitionImplTest.java:1]: Done

File: backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/LifeClassImpl.java (hunk 3 above)
Comment (adaussy) [reply backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/LifeClassImpl.java:34]: Done

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**